recipe/patches/0004-patch-zoneinfo_dir_override-to-point-to-our-tzdata.patch

```diff
@@ -0,0 +1,103 @@
+From 17b067b2a402feaef11eaf2d9fac76d6ba7bd5f6 Mon Sep 17 00:00:00 2001
+From: "H. Vetinari" <h.vetinari@gmx.com>
+Date: Fri, 21 Jun 2024 12:41:07 +1100
+Subject: [PATCH 4/4] patch zoneinfo_dir_override to point to our tzdata
+
+---
+ libstdc++-v3/src/c++20/tzdb.cc | 67 ++++++++++++++++++++++++++++++----
+ 1 file changed, 59 insertions(+), 8 deletions(-)
+
+diff --git a/libstdc++-v3/src/c++20/tzdb.cc b/libstdc++-v3/src/c++20/tzdb.cc
+index c7c7cc9de..2731bb827 100644
+--- a/libstdc++-v3/src/c++20/tzdb.cc
++++ b/libstdc++-v3/src/c++20/tzdb.cc
+@@ -37,8 +37,12 @@
+ #include <mutex>      // mutex
+ #include <filesystem> // filesystem::read_symlink
+ 
+-#ifndef _AIX
+-# include <cstdlib>   // getenv
++#if defined(__linux__)
++# include <dlfcn.h>   // dladdr
++# include <cstdlib>   // free, malloc, memcpy, realpath
++# include <cstring>   // strlen
++#else defined(_WIN32)
++# include <windows.h>
+ #endif
+ 
+ #if defined __GTHREADS && ATOMIC_POINTER_LOCK_FREE == 2
+@@ -68,13 +72,61 @@ namespace __gnu_cxx
+   // Cannot override weak symbols on AIX.
+   const char* (*zoneinfo_dir_override)() = nullptr;
+ #else
+-  [[gnu::weak]] const char* zoneinfo_dir_override();
+-
+-#if defined(__APPLE__) || defined(__hpux__) \
+-  || (defined(__VXWORKS__) && !defined(__RTP__))
+-  // Need a weak definition for Mach-O et al.
+   [[gnu::weak]] const char* zoneinfo_dir_override()
+   {
++    // get path to library we're in, to determine our location relative to $PREFIX;
++    // with help from the MIT-licensed https://github.com/gpakosz/whereami
++    void* addr = __builtin_extract_return_addr(__builtin_return_address(0));
++    char* this_lib;
++#ifdef _WIN32
++    // we're in %PREFIX%\Library\bin\libstdc++-6.dll
++#   define TO_PREFIX "/../.."
++    // needs single quotes for character (not string) literal
++#   define PATH_SEP '\\'
++    wchar_t buffer[MAX_PATH];
++    HMODULE hm = NULL;
++    // non-zero return means success
++    if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
++        GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
++        (LPCSTR) addr, &hm)) {
++      // returns length of string (not counting null byte), see
++      // https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamew#return-value
++      DWORD total_length = GetModuleFileNameW(hm, buffer, sizeof(buffer))
++      if (total_length) {
++        this_lib = (char*)malloc(total_length + 1);
++        memcpy(this_lib, buffer, total_length);
++#else
++    // we're in $PREFIX/lib/libstdcxx.so
++#   define TO_PREFIX "/.."
++#   define PATH_SEP '/'
++    char buffer[PATH_MAX];
```

Comment:
Out of curiosity, where is `PATH_MAX` defined?

Reply:
I took this over from https://github.com/gpakosz/whereami, but some googling says its from `<limits.h>`.